aten/src/ATen/functorch/BatchRulesLinearAlgebra.cpp

```diff
@@ -236,20 +236,16 @@ struct LinalgCheckMatrixBinaryRuleHelper<op_name, F, Func, typelist<A, B, T...>>
       const Tensor& first, std::optional<int64_t> first_bdim,
       const Tensor& second, std::optional<int64_t> second_bdim,
       T... extra_args) {
-    const auto tensor_other = check_inputs_and_reshape_inputs(first, first_bdim, second, second_bdim);
-    const auto tensor_ = std::get<0>(tensor_other);
-    const auto other_ = std::get<1>(tensor_other);
-    return std::make_tuple(Func(tensor_, other_, std::forward<T>(extra_args)...), 0);
+    auto [tensor_, other_]= check_inputs_and_reshape_inputs(first, first_bdim, second, second_bdim);
+    return std::make_tuple(Func(std::move(tensor_), std::move(other_), std::forward<T>(extra_args)...), 0);
   }
 
   static twoOutputs apply_two(
       const Tensor& first, std::optional<int64_t> first_bdim,
       const Tensor& second, std::optional<int64_t> second_bdim,
       T... extra_args) {
-    const auto tensor_other = check_inputs_and_reshape_inputs(first, first_bdim, second, second_bdim);
-    const auto tensor_ = std::get<0>(tensor_other);
-    const auto other_ = std::get<1>(tensor_other);
-    const auto res = Func(tensor_, other_, std::forward<T>(extra_args)...);
+    auto [tensor_, other_]= check_inputs_and_reshape_inputs(first, first_bdim, second, second_bdim);
+    const auto res = Func(std::move(tensor_), std::move(other_), std::forward<T>(extra_args)...);
```

Comment:
Don't both to move the res here on the next line?

Reply:
Fixed